src/backend/gpopt/utils/COptTasks.cpp

```diff
@@ -987,13 +987,13 @@ COptTasks::PrintMissingStatsWarning(CMemoryPool *mp, CMDAccessor *md_accessor,
 	if (0 < rel_stats->Size())
 	{
 		int length = NAMEDATALEN * rel_stats->Size() + 200;
-		char msgbuf[length];
+		std::vector<char> msgbuf(length);
 		snprintf(
-			msgbuf, sizeof(msgbuf),
+			msgbuf.data(), msgbuf.size() * sizeof(char),
```

Comment:
Vector already has an element size in template arguments.
```suggestion
			msgbuf.data(), msgbuf.size(),
```

Reply:
No, size return the number of elements currently held by a vector

Reply:
Sorry, forgot that sprintf() is from C.